### PR TITLE
add def get_match_count

### PR DESCRIPTION
**Pull request by nanli1**
need a function to check key message count
Signed-off-by: nanli <nanli@redhat.com>

related pr  https://github.com/autotest/tp-libvirt/pull/4825/ 

```

/usr/bin/python /usr/local/bin/avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 chardev..file_append
 (1/3) type_specific.io-github-autotest-libvirt.chardev.file_append.serial.append_on: PASS (112.97 s)
 (2/3) type_specific.io-github-autotest-libvirt.chardev.file_append.serial.append_off: PASS (130.14 s)
 (3/3) type_specific.io-github-autotest-libvirt.chardev.file_append.serial.append_default: PASS (129.61 s)

```

### CODE REVIEW

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -312,6 +312,28 @@ def find_substring(string, pattern1, pattern2=None):
     return ret[0]
 
 
+def get_match_count(check_path, key_message):
```
Comment (dzhengfy) [inline virttest/utils_misc.py:315]: check_path -> `file_path` is clearer, I think.

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -312,6 +312,28 @@ def find_substring(string, pattern1, pattern2=None):
...
+    """
+    Get expected messages count in path
+
+    :param check_path: file path to be checked
+    :param key_message: key message that needs to be captured
+    :return count: the count of key message
+    """
+    count = 0
+    try:
+        with open(check_path, 'r', encoding='ISO-8859-1') as fp:
```
Comment (dzhengfy) [inline virttest/utils_misc.py:325]: Add encoding to the function parameter for extensible usage.

File: virttest/utils_misc.py (hunk 1 above)
Comment (dzhengfy) [inline virttest/utils_misc.py:315]: Put this function into utils_logfile.py, thanks.

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: virttest/utils_misc.py (hunk 1 above)
Comment (nanli1) [reply virttest/utils_misc.py]: THanks done

File: virttest/utils_misc.py (hunk 2 above)
Comment (nanli1) [reply virttest/utils_misc.py]: THanks done

File: virttest/utils_misc.py (hunk 1 above)
Comment (nanli1) [reply virttest/utils_misc.py]: THanks done

**Review by dzhengfy — APPROVED** [review]

**Review by Yingshun — APPROVED** [review]

**Pull request merged**